includes/api/class-edd-api.php

```diff
@@ -2260,12 +2301,37 @@ private function get_default_sales_stats() {
 	 */
 	private function get_default_earnings_stats( $args ) {
 
-		// Default earnings return
-		$earnings                              = array();
-		$earnings['earnings']['today']         = $this->stats->get_earnings( 0, 'today', null, $args['include_tax'] );
-		$earnings['earnings']['current_month'] = $this->stats->get_earnings( 0, 'this_month', null, $args['include_tax'] );
-		$earnings['earnings']['last_month']    = $this->stats->get_earnings( 0, 'last_month', null, $args['include_tax'] );
-		$earnings['earnings']['totals']        = edd_get_total_earnings( $args['include_tax'] );
+		$stats  = new EDD\Stats(
+			array(
+				'range'         => 'today',
+				'exclude_taxes' => ! $args['include_tax'],
+				'revenue_type'  => 'net',
+				'output'        => 'typed',
+			)
+		);
+		$earnings['earnings']['today'] = $stats->get_order_earnings();
+
+		$stats  = new EDD\Stats(
+			array(
+				'range'         => 'this_month',
+				'exclude_taxes' => ! $args['include_tax'],
+				'revenue_type'  => 'net',
+				'output'        => 'typed',
+			)
+		);
+		$earnings['earnings']['current_month'] = $stats->get_order_earnings();
+
+		$stats  = new EDD\Stats(
+			array(
+				'range'         => 'last_month',
+				'exclude_taxes' => ! $args['include_tax'],
+				'revenue_type'  => 'net',
+				'output'        => 'typed',
+			)
+		);
+		$earnings['earnings']['last_month'] = $stats->get_order_earnings();
+
+		$earnings['earnings']['totals'] = edd_get_total_earnings( $args['include_tax'] );
```

Comment:
Since this is an independent function, I think it needs to be run through `edd_format_amount` as well in order to get the proper format, or else maybe the `edd_get_total_earnings` function should be run through that instead of what it is now. Anyway, the `totals` for me is still showing too many decimal places, although everything else looks correct now.

Reply:
@robincornett I'm not sure how...b/c that function is run through here:
`return apply_filters( 'edd_total_earnings', round( $total, edd_currency_decimal_filter() ) );`

So it should only ever respond in 2 decimals. Or it is cached for some reason with more decimals since that uses a transient still.

Reply:
I think `edd_get_total_earnings` should be updated to return its value using `edd_format_amount`. I'm getting inconsistent results with `round` even though `edd_currency_decimal_filter()` is returning 2.

For example, today, the `totals` in the API request is showing as `5645`, which is exactly what the transient is. The current code is not formatting the amount to include the decimal places, but updating the function to use `edd_format_amount` instead properly displays `"5645.00"`.

I manually updated the transient to match what it was previously: `5336.8`. The `totals` amount is now showing as `5336.8000000000002` instead of the expected `"5336.80"`.

Reply:
I think updating `edd_get_total_earnings` has far more reaching side affects, as your data type changed to a string from a float. If we just make the API format the totals using the `maybe_format_amount` method we should be in the clear.

this would affect the API data only, and not bring changes to the other areas of EDD calling `edd_get_total_earnings`.

Reply:
I'll play around with this and commit when I have something I'm comfortable with for this.